server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -962,6 +969,60 @@ public boolean isForceExecution() {
         );
     }
 
+    static boolean shouldStoreFailure(String indexName, Metadata metadata, long epochMillis) {
```

Comment:
These methods could probably all have javadoc, so we can indicate what they're intended to do (and to help IDEs)